### ISSUE TITLE
Server Side environment variables not working

### ISSUE THREAD

**Issue by timsawtellgo1** [expatfile/next-runtime-env#35]
When following the instructions in the example app's README.md, the server side environment variables are undefined.

```
cd examples/with-app-router
yarn build
NEXT_PUBLIC_FOO=foo-value BAR=bar-value yarn start
```

and visiting `http://localhost:3000/server-side` I see

![image](https://github.com/expatfile/next-runtime-env/assets/99709659/5268d392-0660-4059-b02d-27a573d15d8f)

however visiting `http://localhost:3000/client-side` I see

![image](https://github.com/expatfile/next-runtime-env/assets/99709659/66f02e0f-f3f4-4883-b593-f02ed145a21b)

I tried a few things, and the only way I could get the server side version to work was to set `NODE_ENV=development` when using `yarn start`.

after running `NEXT_PUBLIC_FOO=foo-value BAR=bar-value NODE_ENV=development yarn start` I see 
![image](https://github.com/expatfile/next-runtime-env/assets/99709659/877ac37b-7e00-4246-847d-eea97b803896)

This is not going to work in production because we need our NextJS app to run in production mode. It seems like the only way this library works as stated (identical server + client side environment variables) is if we run the app in development mode. Am I missing something?

Thanks heaps for this library by the way, gobsmacked when I learned of NextJS's approach to environment variables and this seems like a great way to solve it - I'm just unable to get server side env variables working in production mode though.

**Comment by loba-b** [comment]
Unfortunately, I have the same findings

**Comment by HofmannZ** [comment]
@timsawtellgo1 - Thanks for reporting this. We're currently working on a production-ready solution for using the app dir. The problem with the current example is that server components are statically generated by default at build time. To ensure the env vars are loaded, the server component needs to be forced to be dynamic.

I've updated the example. Can you give it a spin? :)

Additionally, I've added an issue for tracing the overall app router support. See: #69

**Comment by HofmannZ** [comment]
Hi 👋

(It's been a while, I know)

We've completely revamped our solution to support Server Components and the App Router. 🙌

Please check the [development](https://github.com/expatfile/next-runtime-env/tree/development) branch for docs and examples.

We've also published a version [2.0.0-alpha.1 on NPM](https://www.npmjs.com/package/next-runtime-env/v/2.0.0-alpha.1). 🚀

**Comment by branostupak** [comment]
Hi 👋 What is the stage of the server side runtime env variables in current 1.7.4? We're following documented example, but env variables do not work server-side. Accessing env variables through `process.env.` makes them undefined.
